app/src/main/java/com/swiezaki/swiezakimobile/network/RxErrorHandlingCallAdapterFactory.kt

```diff
@@ -0,0 +1,46 @@
+package com.swiezaki.swiezakimobile.network
+
+import com.swiezaki.swiezakimobile.network.error.*
+import io.reactivex.Observable
+import okhttp3.ResponseBody
+import retrofit2.Call
+import retrofit2.CallAdapter
+import retrofit2.HttpException
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import java.io.IOException
+import java.lang.reflect.Type
+
+class RxErrorHandlingCallAdapterFactory(private val original: RxJava2CallAdapterFactory) : CallAdapter.Factory() {
+
+    override fun get(returnType: Type, annotations: Array<out Annotation>, retrofit: Retrofit): CallAdapter<*, Observable<*>> {
+        return RxCallAdapter(retrofit, original.get(returnType, annotations, retrofit))
+    }
+}
+
+class RxCallAdapter<R>(private val retrofit: Retrofit, private val original: CallAdapter<R, *>?) : CallAdapter<R, Observable<*>> {
+
+    override fun adapt(call: Call<R>): Observable<*> {
+        return (original?.adapt(call) as Observable<*>).onErrorResumeNext { throwable: Throwable -> Observable.error(asRetrofitException(throwable)) }
```

Comment:
divide that into two lines at least

Reply:
Oh, onErrorResumeNext should be in next line, I didn't notice that